ota-plus-web/app/com/advancedtelematic/auth/oidc/NamespaceProvider.scala

```diff
@@ -22,19 +24,33 @@ class NamespaceFromIdentity extends NamespaceProvider {
 class NamespaceFromUserProfile @Inject()(val conf: Configuration,
                                          val ws: WSClient,
                                          val clientExec: ApiClientExec,
+                                         val oidcGateway: OidcGateway,
                                          val tracer: ZipkinTraceServiceLike)
                                         (implicit ec: ExecutionContext)
   extends NamespaceProvider with ApiClientSupport {
 
+  implicit private val traceData = TraceData(tracer.tracing.tracer().newTrace())
+
+  private val namespaceFromJson: JsValue => Namespace = json => Namespace((json \ "defaultNamespace").as[String])
+
+  private def createUser(accessToken: AccessToken, userId: UserId) =
+    oidcGateway
+      .getUserInfo(accessToken)
+      .flatMap { ic =>
+        userProfileApi
+          .createUser(userId, ic.name, ic.email)
+          .map(namespaceFromJson)
+      }
+
   override def apply(tokens: Tokens): Future[Namespace] = {
-    implicit val traceData = TraceData(tracer.tracing.tracer().newTrace())
-
-    userProfileApi.userOrganizations(tokens.idToken.userId).map {
-      case organizations if organizations.isEmpty =>
-        Namespace.generate
-      case organizations =>
-        organizations.filter(_.isDefault).map(_.namespace).head
-    }
+    val userId = tokens.idToken.userId
+    userProfileApi
+      .getUser(userId)
+      .map(namespaceFromJson)
+      .recoverWith {
+        case RemoteApiError(result, _) if result.header.status == 404 =>
```

Comment:
Is there any chance user profile also returns an ErrorRepresentation and we can see that 404 is not a broken routing or something but a specific response for a non existing user?

Reply:
I found a way to check for that but I'm not too happy with it, see my last commit. I don't think there is an easy way to check for a specific remote API error like that without doing some refactoring around our API clients...
So in my opinion I'd leave that out of this PR. How likely is that route would be broken? I'd venture not much...

Reply:
I'll merge this for now and open a new PR with that last commit if we want it :)

Reply:
@koshelev see this one for the commit I was talking about https://github.com/advancedtelematic/ota-plus-server/pull/2094.